lambdas/service/app.py

```diff
@@ -546,204 +546,244 @@ def repository_search(entity_type: str, item_id: str):
+                        `search_before`.
+                    '''),
+            params.query(
+                'search_after_uid',
+                schema.optional(str),
+                description='''
+                        The value of the ElasticSearch UID corresponding to the hit
+                        above, if `search_after` is set.
+                    '''),
+            params.query(
+                'search_before',
+                schema.optional(str),
+                description='''
+                        The value of the \'sort\' field for the hit after which all
+                        results should be returned. Mutually exclusive with
+                        `search_after`.
+                    '''),
+            params.query(
+                'search_before_uid',
+                schema.optional(str),
+                description='''
+                        The value of the ElasticSearch UID corresponding to the hit
+                        above, if `search_before` is set.
+                    ''')
+        ],
+        'summary': f'Search the \'{entity_type}s\' index for entities of interest.',
+        'responses': {
+            '200': {
+                'description': format_description(f'''
+                    Paginated list of {entity_type}s that meet the search
+                    criteria ("hits"). The structure of these hits is documented
+                    under the [corresponding endpoint for a specific entity]\
+(#operations-Repository-get_repository_{entity_type}s__{entity_type}_id_).
+
+                    The `pagination` sections describes the total number of hits
+                    and total number of pages, as well as user-supplied search
+                    parameters such as page size and sorting behavior. It also
+                    provides as links for navigating forwards and backwards
+                    between pages of results.
+
+                    The `termFacets` section tabulates the occurrence of unique
+                    values within nested fields of the `hits` section. Not every
+                    nested field is tabulated, but the set of tabulated fields
+                    is consistent between entity types.
+                '''),
+                'content': {
+                    'application/json': {
+                        'schema': page_schema
+                    }
+                }
+            }
+        }
+    }
+
+
+def repository_id_spec(entity_type: str):
+    return {
+        'tags': ['Repository'],
+        'summary': f'Detailed information on a particular {entity_type} entity.',
+        'parameters': [
+            params.path(f'{entity_type}_id', str, description=f'The UUID of the desired {entity_type}')
+        ],
+        'responses': {
+            '200': {
+                'description': format_description(f'''
+                    This response describes a single {entity_type}, possibly at
```

Comment:
Weasel words. Does it or does it not have additional detail? I don't think it does.

Reply:
I found some instances in which it does. Compare 

` http 'https://service.dev.explore.data.humancellatlas.org/repository/projects?filters={"entryId":{"is":["efb89bf7-5191-4dd2-8fe6-c6cb15f38dce"]}}' | jq .hits[].projects`

With

`http 'https://service.dev.explore.data.humancellatlas.org/repository/projects/efb89bf7-5191-4dd2-8fe6-c6cb15f38dce' | jq .projects`

Reply:
I forgot about that hack. This quirk should be documented for the /projects endpoint specifically.

Reply:
Or even better, the hack could be eliminated. I don't think it would incur a big performance penalty if those additional project properties were present on each hit of the regular `/projects` response.

Reply:
Could you quickly gauge the performance impact? If it is negligible, file a ticket to add that behaviour and leave the quirk undocumented. Otherwise, document the quirk.